src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -11306,6 +11306,29 @@ instruct convD2L_reg_reg(rRegL dst, regD src, rFlagsReg cr)
   ins_pipe(pipe_slow);
 %}
 
+instruct convF2HF_reg_reg(rRegI dst, regF src, regF tmp) %{
+  effect(TEMP tmp);
+  match(Set dst (ConvF2HF src));
+  format %{ "vcvtps2ph $dst,$src \t using $tmp as TEMP"%}
+  ins_encode %{
+     __ vcvtps2ph($tmp$$XMMRegister, $src$$XMMRegister, 0x04, Assembler::AVX_128bit);
+     __ movdl($dst$$Register, $tmp$$XMMRegister);
+     __ movswl($dst$$Register, $dst$$Register);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct convHF2F_reg_reg(regF dst, rRegI src) %{
+  match(Set dst (ConvHF2F src));
+  format %{ "vcvtph2ps $dst,$src" %}
+  ins_encode %{
+     __ movdl($dst$$XMMRegister, $src$$Register);
+     __ vcvtph2ps($dst$$XMMRegister, $dst$$XMMRegister, Assembler::AVX_128bit);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
```

Comment:
For HF2F, good to also add optimized rule with LoadS to benefit from vcvtph2ps memory src form of instruction.
match(Set dst (ConvHF2F ( LoadS mem)));

Reply:
Hi @sviswa7 , with this rule, I dont see additional performance gain. So I have not added it in the latest update.